spec/cable/handler_spec.cr

```diff
@@ -59,9 +59,30 @@ describe Cable::Handler do
         seq += 1
         ws2.close if seq >= messages.size
       end
+      ws2.on_close do |code, _reason|
+        code.should eq(HTTP::WebSocket::CloseCode::AbnormalClosure)
+      end
       ws2.send({"command" => "subscribe", "identifier" => {channel: "ChatChannel", room: "1"}.to_json}.to_json)
 
       ws2.run
+
+      # should be 1 connection open
+      Cable.server.connections.size.should eq(1)
+    end
+
+    it "rejected" do
+      address_chan = start_server
+      listen_address = address_chan.receive
+
+      ws2 = HTTP::WebSocket.new("ws://#{listen_address}/updates?test_token=reject")
+      ws2.on_close do |code, reason|
+        code.should eq(HTTP::WebSocket::CloseCode::NormalClosure)
+        reason.should eq("Farewell")
+      end
+      ws2.run
+
+      # should be zero connections open
+      Cable.server.connections.size.should eq(0)
```

Comment:
I get this error locally on this line when running the specs:

> Error writing to socket: Broken pipe (IO::Error)

Reply:
Super weird as its working fine on my local machine, although I'm using 1.5.1. I'll dig into it and see whats going on

Reply:
The mock client code was raising the exception, so I just wrapped that around a rescue block to ensure that it doesn't get in the way of the handler tests. 
I found another weird behaviour while trying to debug it. Which we now account for, see my comments in the handler on_message changes.